cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -150,15 +158,13 @@ public Boolean call() throws Exception {
 
 
   private void checkServicesWithRetry(Callable<Boolean> callable,
-                                      String exceptionMessage) throws TimeoutException {
-    int numSecs = 0;
+                                      String exceptionMessage) throws TimeoutException, InterruptedException {
```

Comment:
I moved `TimeUnit.Seconds.Sleep(1)` outside of the try block, and it also throws the `InterruptedException`.
I moved it down so that even if the callable throws an exception, the sleep shouldn't be skipped.